### PR TITLE
Reenable and fix tests in ContentNegotiationSpec

### PR DESCRIPTION
**Pull request by aruediger**
also make it more agnostic to newline encodings

/cc @jrudolph

### CODE REVIEW

**Comment by 2m** [comment]
OK TO TEST

File: akka-http-tests/src/test/scala/akka/http/scaladsl/marshalling/ContentNegotiationSpec.scala [hunk 1]
```diff
@@ -106,8 +106,8 @@ class ContentNegotiationSpec extends FreeSpec with Matchers {
       accept(`text/plain`, `text/html`) should select(`text/html`, `UTF-8`)
     }
 
-    """Accept: text/html, text/plain;q=0.8, application/*;q=.5, *;q= .2
-       Accept-Charset: UTF-16""" test { accept ⇒
+    """Accept: text/html, text/plain;q=0.8, application/*;q=.5, */*;q= .2
```
Comment (2beaucoup) [inline akka-http-tests/src/test/scala/akka/http/scaladsl/marshalling/ContentNegotiationSpec.scala]: @jrudolph I wonder how #17987 did pass this test? (`* -> */*`)

Comment (2beaucoup) [inline akka-http-tests/src/test/scala/akka/http/scaladsl/marshalling/ContentNegotiationSpec.scala]: ignore plz. the newline caused the failure.

**Comment by 2beaucoup** [comment]
What exactly failed here? I can't access Jenkins since port != 80. :(

**Comment by 2m** [comment]
There was a failure in `TcpSpec`. Probably unrelated to your changes.
```
[info] - must shut down properly even if some accepted connection Flows have not been subscribed to *** FAILED *** (3 seconds, 27 milliseconds)
[info]   java.util.concurrent.TimeoutException: Futures timed out after [3 seconds]
[info]   at scala.concurrent.impl.Promise$DefaultPromise.ready(Promise.scala:219)
[info]   at scala.concurrent.impl.Promise$DefaultPromise.result(Promise.scala:223)
[info]   at scala.concurrent.Await$$anonfun$result$1.apply(package.scala:107)
[info]   at scala.concurrent.BlockContext$DefaultBlockContext$.blockOn(BlockContext.scala:53)
[info]   at scala.concurrent.Await$.result(package.scala:107)
[info]   at akka.stream.io.TcpSpec$$anonfun$2$$anonfun$apply$mcV$sp$34$$anonfun$apply$mcV$sp$35.apply$mcV$sp(TcpSpec.scala:522)
[info]   at akka.stream.io.TcpSpec$$anonfun$2$$anonfun$apply$mcV$sp$34$$anonfun$apply$mcV$sp$35.apply(TcpSpec.scala:508)
[info]   at akka.stream.io.TcpSpec$$anonfun$2$$anonfun$apply$mcV$sp$34$$anonfun$apply$mcV$sp$35.apply(TcpSpec.scala:508)
[info]   at akka.stream.testkit.Utils$.assertAllStagesStopped(Utils.scala:26)
[info]   at akka.stream.io.TcpSpec$$anonfun$2$$anonfun$apply$mcV$sp$34.apply$mcV$sp(TcpSpec.scala:508)
[info]   at akka.stream.io.TcpSpec$$anonfun$2$$anonfun$apply$mcV$sp$34.apply(TcpSpec.scala:508)
[info]   at akka.stream.io.TcpSpec$$anonfun$2$$anonfun$apply$mcV$sp$34.apply(TcpSpec.scala:508)
[info]   at org.scalatest.Transformer$$anonfun$apply$1.apply$mcV$sp(Transformer.scala:22)
...
```

**Comment by 2beaucoup** [comment]
Can we merge this? Failure looks unrelated.

File: akka-http-core/src/test/scala/akka/http/impl/engine/ws/BitBuilder.scala [hunk 2]
```diff
@@ -13,10 +13,9 @@ import parboiled2._
 object BitBuilder {
   implicit class BitBuilderContext(val ctx: StringContext) {
     def b(args: Any*): ByteString = {
-      val parser = new BitSpecParser(ctx.parts.mkString)
+      val input = ctx.parts.mkString.replace("\r\n", "\n")
```
Comment (jrudolph) [inline akka-http-core/src/test/scala/akka/http/impl/engine/ws/BitBuilder.scala]: Uhoh. Thanks for that one.

**Comment by jrudolph** [comment]
Thanks, @2beaucoup. LGTM.

**Comment by 2beaucoup** [comment]
rebased

**Comment by ktoso** [comment]
LGTM, thanks for rebasing @2beaucoup :)

**Pull request merged**